eo-maven-plugin/src/test/java/integration/MojasIT.java

```diff
@@ -0,0 +1,197 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package integration;
+
+import com.yegor256.Mktmp;
+import com.yegor256.MktmpResolver;
+import com.yegor256.WeAreOnline;
+import com.yegor256.farea.Farea;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import org.eolang.maven.AssembleMojo;
+import org.eolang.maven.ContainsFiles;
+import org.eolang.maven.ParseMojo;
+import org.eolang.maven.PullMojo;
+import org.eolang.maven.ResolveMojo;
+import org.eolang.maven.ShakeMojo;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Integration tests for mojas.
+ *
+ * @since 0.52
+ */
+@SuppressWarnings({"JTCOP.RuleAllTestsHaveProductionClass", "JTCOP.RuleNotContainsTestWord"})
+@ExtendWith(WeAreOnline.class)
+@ExtendWith(MktmpResolver.class)
+final class MojasIT {
+
+    @Test
+    void assemblesTogether(@Mktmp final Path temp) throws IOException {
+        final String stdout = "target/eo/%s/org/eolang/io/stdout.%s";
+        final String parsed = String.format(stdout, ParseMojo.DIR, AssembleMojo.XMIR);
+        final String optimized = String.format(stdout, ShakeMojo.DIR, AssembleMojo.XMIR);
+        final String pulled = String.format(stdout, PullMojo.DIR, AssembleMojo.EO);
+        new Farea(temp).together(
+            f -> {
+                f.clean();
+                f.dependencies()
+                    .append("org.eolang", "eo-runtime", "0.51.6");
+                f.files()
+                    .file("src/main/eo/one/main.eo")
+                    .write(MojasIT.helloWorld().getBytes(StandardCharsets.UTF_8));
+                f.build()
+                    .plugins()
+                    .appendItself()
+                    .execution("tests")
+                    .goals("register", "assemble");
+                f.exec("package");
+                MatcherAssert.assertThat(
+                    String.join(
+                        " ",
+                        "AssembleMojo should have placed runtime",
+                        "library, but didn't"
+                    ),
+                    f.files().file(
+                        String.format(
+                            "target/eo/%s/org.eolang/eo-runtime/-/0.51.6/org/eolang/Phi.class",
+                            ResolveMojo.DIR
+                        )
+                    ).exists(),
+                    Matchers.is(true)
+                );
+            }
+        );
+        MatcherAssert.assertThat(
```

Comment:
@Suban05 let's either make assertions inside `Farea` lambda like you did 15 lines above or outside of it